src/main/scala/dpla/ingestion3/enrichments/SpatialEnrichment.scala

```diff
@@ -0,0 +1,275 @@
+  private def county(j: JValue): Option[String] = {
+    if (woeType(j) == Some(9)) shortName(j)
+    else parentFeatureName(j, 9)
+  }
+  private def state(j: JValue): Option[String] = {
+    if (woeType(j) == Some(8)) shortName(j)
+    else parentFeatureName(j, 8)
+  }
+  private def country(j: JValue): Option[String] = {
+    if (woeType(j) == Some(12)) shortName(j)
+    else parentFeatureName(j, 12)
+  }
+
+  // Region is implemented inconsistently in the Ingestion1 Twofishes enricher,
+  // and Twofishes does not return it consistently. If you look up "United
+  // States" it will provide the continent, "North America", but if you look up
+  // a city in the United States it will not return the continent.  Filling in
+  // a region value will provide misleading searches.
+  // private def region(j: JValue): String = ???
+
+  /**
+    * Return a single String of the Place's coordinates, joined by ","
+    *
+    * @param j  Twofishes response JValue
+    * @return   The coordinates as at "lat, lng" string
+    */
+  private def coordinates(j: JValue): Option[String] = {
+    val c = ((((j \ "interpretations")(0) \ "feature") \ "geometry") \
+      "center").extract[Coords]
+    Option(c.lat.toString + "," + c.lng.toString)
+  }
+
+  /**
+    * Return the place name or None if it's the place's coordinates
+    *
+    * If the place's name field has been used to convey coordinates, we do want
+    * to override this when we fill it in.  Otherwise, we don't want to clobber
+    * it if it's already defined.
+    *
+    * @param nameCandidate
+    * @param j
+    */
+  private def nameUnlessCoords(nameCandidate: Option[String], j: JValue):
+      Option[String] = {
+    nameCandidate match {
+      case s if s.isDefined => {
+        if (s.mkString matches """[\+\-\d\.]+\s*,\s*[\+\-\d\.]+""") {
+          // It's coordinates, so return the name instead
+          displayName(j)
+        } else {
+          s
+        }
+      }
+      case _ => None
+    }
+  }
+
+  /**
+    * Enrich the given DplaPlace
+    *
+    * @param p  The Place
+    * @return   An enriched copy of the Place
+    */
+  def enrich(p: DplaPlace): DplaPlace = {
+    try {
+      queryTerm(p) match {
+        case Some(term) => {
+          val j = geocoder.geocoderResponse(term)
+          // Return the looked-up place, unless it was a country.
+          if (woeType(j) != Some(12)) {
+            p.copy(
+              name = nameUnlessCoords(p.name, j),
+              city = p.city.orElse(city(j)),
+              county = p.county.orElse(county(j)),
+              state = p.state.orElse(state(j)),
+              country = p.country.orElse(country(j)),
+              coordinates = p.coordinates.orElse(coordinates(j))
+            )
+          } else {
+            p
+          }
+        }
+        case _ => p
+      }
+    } catch {
+      case e: Exception => {
+        logger.error(e.toString)  // TODO: Catch this higher up and store it?
```

Comment:
Do we know what types of errors might happen? I think we want to distinguish the cases of "twofishes blew up" from the cases of "couldn't find a result for x". And we also don't want to have to check the logs to find out that our output is incorrect because of a network problem. I'd argue for actually letting the exception bubble up here.